### PR TITLE
01867: EET - Enhance HapiGetTxnRecord

### LINKED ISSUE #1844: Enhance EET to validate transaction record with entity matching
### Problem

Currently, the transactionRecord is printed out to manually validate the token Transfer lists and assessed custom fee.

### Solution

Add validators to `HapiGetTxnRecord` to validate the entities by matching in the appropriate transferLists.

### Alternatives

_No response_

### LINKED ISSUE #1867: Enhance test-clients HapiGetTxnRecord
### Problem

Needs of validating `accountAmounts`, token account amounts, `nftTransfers`, `assessed_custom_fees` and its size in test-clients `HapiGetTxnRecord` for HIP-17 and HIP-18.

### Solution

Add the following needed validations:
- [x] `accountAmounts`
- [x] token account amounts
- [x] `nftTransfers`
- [x] `assessed_custom_fees`
- [x] the size of `assessed_custom_fees`

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by qnswirlds**
**Related issue(s)**:

- Closes #1844 
- Closes #1867

**Notes for reviewer**:

- [x] Added the following needed validations in test-clients `HapiGetTxnRecord` for HIP-17 and HIP-18
- - [x] `accountAmounts`
- - [x] token account amounts
- - [x] `nftTransfers`
- - [x] `assessed_custom_fees`
- - [x] the size of `assessed_custom_fees`

- [x] Added validations to all HIP-18 related specs in the `TokenTransactSpecs`, some of them also touch NFTs.

### CODE REVIEW

**Review by abhishek-hedera — DISMISSED** [review]
LGTM

**Review by anighanta — APPROVED** [review]
LGTM. 
Just one change required I think. Merging master will fix the test that is toggled in `RecordCreationSuite`

**Review by qnswirlds — COMMENTED** [review]
already done here @anighanta

File: test-clients/src/main/java/com/hedera/services/bdd/suites/records/RecordCreationSuite.java [hunk 1]
```diff
@@ -92,7 +92,7 @@ public static void main(String... args) {
 	protected List<HapiApiSpec> getSpecsInSuite() {
 		return List.of(
 				new HapiApiSpec[] {
-//						confirmNftToggleIsOffByDefaultThenEnable(),
+						confirmNftToggleIsWorksThenReenable(),
```
Comment (qnswirlds) [inline test-clients/src/main/java/com/hedera/services/bdd/suites/records/RecordCreationSuite.java]: sorry, somehow it didn't get my selection. Here @anighanta

**Pull request merged**